fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientLifecycleEvents.java

```diff
@@ -51,13 +52,39 @@ private ClientLifecycleEvents() {
 		}
 	});
 
+	/**
+	 * Called when the current player in the Minecraft game has moved to a different world.
+	 * This occurs when a new world has been created, but before the client's {@link MinecraftClient#world current world} is disposed.
+	 */
+	public static final Event<ChangeWorld> CHANGE_WORLD = EventFactory.createArrayBacked(ChangeWorld.class, callbacks -> (client, world, destination) -> {
+		for (ChangeWorld callback : callbacks) {
+			callback.onChangeWorld(client, world, destination);
+		}
+	});
+
+	@Environment(EnvType.CLIENT)
```

Comment:
If ClientLifecycleEvents is already being stripped isn't this unnecessary?

Reply:
At bytecode level nested/inner classes can still be touched outside of client

Reply:
For the JVM inner classes are just treated as other classes in the same package. So when outer classes are removed, inner classes, like other classes in the same package, are not touched.